### ISSUE TITLE
[Spike] - See if vanity imports puts us in an infinite loop

### ISSUE THREAD

**Issue by marwan-at-work** [gomods/athens#451]
1. Create a vanity import under your own domain name, `marwan.io/mycoolpackage` 
2. Have the vanity import return a meta to an athens server (localhost since we don't have athens up yet) `<meta name="go-import" content="marwan.io/mycoolpackage mod http://localhost:3000">
3. do a go get for this package with GOPROXY set to localhost:3000
4. Proxy will internally call `go mod download` on `marwan.io/mycoolpackage`, which will then redirect `go mod download` to use a mod server, which is the Proxy and so we potentially end up in an infinite loop. 

Whether we redirect to CDN or not matters, but that depends on the type of storage and whether it's Proxy/Registry.

**Comment by robjloranger** [comment]
If there is no other, rjl.li/license exists

**Comment by arschles** [comment]
note that as of #772, we're not going to try and build a registry for the time being. this still applies to the proxy though

**Comment by arschles** [comment]
@marwan-at-work good point (and sorry for the delay). I see what you mean now. What do you think about recording each cache miss that we see and cancelling a `go mod download` call if we see the same cache miss within duration D?

I think this relies on #760 being built, but assuming it is, I can't think of any other reason that there would be multiple cache misses within a small duration for the same module@version
